3rdparty/CMakeLists.txt

```diff
@@ -90,8 +90,18 @@ if (NBL_COMPILE_WITH_SDL2)
 	add_subdirectory(SDL2 SDL2 EXCLUDE_FROM_ALL)
 endif()
 
+find_package(OpenGL OPTIONAL_COMPONENTS EGL)
+find_package(Wayland)
+#[[if (OpenGL_EGL_FOUND)
+	message(FATAL_ERROR "FOUND EGL!!")
+else()
+	message(FATAL_ERROR "COULDNT FIND EGL!!")
+endif()]]
 # EGL
 set(EGL_NO_GLEW ON)
+if (OpenGL_EGL_FOUND AND WAYLAND_CLIENT_FOUND AND UNIX AND NOT APPLE)
+	set(EGL_UNIX_USE_WAYLAND ON)
+endif()
```

Comment:
why is this a config thing, weren't we supposed to do this all dynamically?

Reply:
yeah but in case of Wayland it's pretty fat header-wise so idk

Reply:
just find the system headers only... if you dont find refuse to build

(if for some strange reason the user cannot install `libwayland-dev` even if they want to run X11 everyday, then they could just set the wayland header directory variable manyally in Cmake and make our code proceed)